platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -409,12 +409,24 @@ class ConfigurationCacheProblems(
     }
 
     private
-    fun areDegradationReasonsPresent(): Boolean = degradationController.degradationReasons.isNotEmpty()
+    fun areDegradationReasonsPresent(): Boolean = degradationController.hasDegradationReasons
+
+    private
+    fun areTaskDegradationReasonsPresent(): Boolean = degradationController.hasTaskDegradationReasons
 
     private
     fun degradationSummary(): String {
-        val degradingTasks = degradationController.degradationReasons.keys
-        return " because incompatible ${if (degradingTasks.size > 1) "tasks were" else "task was"} found."
+        val degradingTasks = degradationController.taskDegradationReasons.size
+        val degradingFeatures = degradationController.featureDegradationReasons
```

Comment:
💅 Why `degradingTasks` is a size of the collection and `degradingFeatures` is a collection itself? Can we be more consistent here either with names or with values?